perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/userstabs/SelfApplicationDetailTabItem.java

```diff
@@ -115,10 +116,26 @@ public Widget draw() {
 		vp.add(sp);
 		vp.setCellHeight(sp, "100%");
 
-		GetApplicationDataById data = new GetApplicationDataById(applicationId);
-		data.setShowAdminItems(false);
-		data.retrieveData();
-		sp.setWidget(data.getContents());
+		if (application.getType().equalsIgnoreCase("EMBEDDED")) {
+			final FlexTable ft = new FlexTable();
+			ft.setCellSpacing(5);
+			ft.setHTML(0, 0, "<strong>Name:</strong> " + SafeHtmlUtils.fromString(application.getUser().getFullNameWithTitles()).asString());
+			new GetEntityById(PerunEntity.RICH_USER_WITH_ATTRS, application.getUser().getId(), new JsonCallbackEvents(){
+				@Override
+				public void onFinished(JavaScriptObject jso) {
+					User user = jso.cast();
+					String mail = SafeHtmlUtils.fromString(user.getAttribute("urn:perun:user:attribute-def:def:preferredMail").getValue()).asString();
```

Comment:
Same here. What will happen when the user does not have a preferred mail set?

Reply:
I have added check here as well, thanks.